google/ggrc-core
# GGRC-7655 Fix performance in Change Log for Global Creator

# Issue description

Currently revisions for Change Log load too slowly for Global Creator user if it has read access to a lots of objects.

# Steps to test the changes

1. Setup dump from QA instance;
1. Login as Administrator;
2. Open any Audit or create a new one;
3. Check that Global Creator user "globcreator@gmail.com" is Audit Captain or Auditor;
4. Login as "globcreator@gmail.com".
6. Open Change Log tab of the Audit from step 2;

**Actual Result:**
Data loading takes ~1 min in the Change Log tab for Global Creator. For Admin it takes 2-3 sec.

**Expected Result:**
Data should be loaded in less amount of time for Global Creator.

# Solution description

The issue with performance was caused by the way permissions for revisions where checked.

Since `ggrc.rbac.permissions` does not handle revisions, all objects current user has "read" permission on should be gathered, and only revisions of such objects could be obtained by FE which is implemented through corresponding SQL filter. But applying such a filter is time consuming operation.

The fix consists of modification of `QueryHelper._get_type_query` method in way that it will parse information about requested revision's resource from query API query and check "read" permissions only for this resources.

# Sanity checklist




# PR Review checklist
